src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,191 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(last, value, null);
+        if (last == null) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> currentNode = findElement(index);
+        checkIndex(index, index > size);
+        if (index == 0) {
+            addFirst(value, currentNode);
+            return;
+        }
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> prev = currentNode.prev;
+            Node<T> newNode = new Node<>(prev, value, currentNode);
+            currentNode.prev = newNode;
+            if (prev == null) {
+                first = newNode;
+            } else {
+                prev.next = newNode;
+            }
+            size++;
+        }
+    }
 
+    private void addFirst(T value, Node<T> currentNode) {
+        Node<T> newNode = new Node<>(null, value, currentNode);
+        first = newNode;
+        if (currentNode == null) {
+            last = newNode;
+        } else {
+            currentNode.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        if (list == null) {
+            return false;
+        }
+        for (T value : list) {
+            add(value);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, index >= size);
+        return findElement(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index, index >= size);
+        Node<T> newNode = findElement(index);
+        T oldValue = newNode.item;
+        newNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index, index >= size);
+        Node<T> currentNode = findElement(index);
+        Node<T> prev = currentNode.prev;
+        Node<T> next = currentNode.next;
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+        }
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+        }
+        size--;
+        return currentNode.item;
```

Comment:
let's replace method delete() with this logic (from line 87 to line 100)
and then we can use method delete() here and in the method remove by value

Reply:
done